### PR TITLE
Feat/environment improvements

### PR DESCRIPTION
**Pull request by cardosofede**
**Before submitting this PR, please make sure**:

- [ ] Your code builds clean without any errors or warnings
- [ ] You are using approved title ("feat/", "fix/", "docs/", "refactor/")

**A description of the changes proposed in the pull request**:
After upgrading the conda version to latest, the environment had some issues with dependencies and it wasn't possible to create the environment.
This PR upgrades the python version to 3.10 and the version of the most important dependencies.

**Tests performed by the developer**:
- Unit test
- Candles example

**Tips for QA testing**:
- Install and compile
- Build docker image multi-arch
- Run a strategy with a real account

### CODE REVIEW

**Review by fengtality — APPROVED** [review]

File: test/hummingbot/connector/derivative/bitget_perpetual/test_bitget_perpetual_order_book_data_source.py [hunk 1]
```diff
@@ -626,7 +626,7 @@ def test_listen_for_order_book_snapshots_log_exception(self, mock_api):
             self.data_source.listen_for_order_book_snapshots(self.ev_loop, msg_queue)
         )
         try:
-            self.async_run_with_timeout(self.listening_task)
+            self.async_run_with_timeout(self.listening_task, timeout=2)
```
Comment (keithbaum) [inline test/hummingbot/connector/derivative/bitget_perpetual/test_bitget_perpetual_order_book_data_source.py:629]: Thanks for fixing this!!!

**Comment by nikspz** [comment]
ongoing issue on fresh installation.
python-telegram-bot related issue at every bot start (WSL Ubuntu source client)
will continue investigation after staging regression tests

```
/home/comp1/miniconda3/envs/hummingbot/lib/python3.10/site-packages/telegram/utils/request.py:47: UserWarning: python-telegram-bot is using upstream urllib3. This is allowed but not supported by python-telegram-bot maintainers.               warnings.warn('python-telegram-bot is using upstream urllib3. This is allowed but not '                               /home/comp1/newpython6389/bin/hummingbot.py:81: DeprecationWarning: There is no current event loop                        ev_loop: asyncio.AbstractEventLoop = asyncio.get_event_loop() 
```
image

**Comment by nikspz** [comment]
Error at the start of Client seems fixed, but before I was able to start I have to reinstall bot 3 times got this issue.

Anaconda3 + python3.10 PR
image

Then I tried clean install 3 times 
on the third one I use `conda clean -a` before installation
and then seems fixed:

image
image

p.s. bit_com_perpetual started showing error requesting exchange info as well on development

File: test/hummingbot/remote_iface/test_mqtt.py [hunk 2]
```diff
@@ -71,6 +69,10 @@ def tearDownClass(cls) -> None:
     def setUp(self) -> None:
         super().setUp()
         self.log_records = []
+        self.original_event_loop = asyncio.get_event_loop()
```
Comment (TheHolyRoger) [inline test/hummingbot/remote_iface/test_mqtt.py:72]: @cardosofede I've reworked these tests in this PR https://github.com/hummingbot/hummingbot/pull/6388/files

can you revert these changes and apply that PR first to see if these still need applying? There shouldn't be a need to create a new event loop.

Comment (cardosofede) [reply test/hummingbot/remote_iface/test_mqtt.py:72]: will do!

**Comment by TheHolyRoger** [comment]
My PR with updated changes is here https://github.com/cardosofede/hummingbot/pull/2

I've updated cython to alpha10 https://cython.readthedocs.io/en/latest/src/changes.html#alpha-10-2022-01-06

I think we should test it as is, it compiles fine for me. But maybe push to an even newer cython release as part of this PR

File: bin/hummingbot.py [hunk 3]
```diff
@@ -78,7 +78,13 @@ async def main_async(client_config_map: ClientConfigAdapter):
 def main():
     chdir_to_data_directory()
     secrets_manager_cls = ETHKeyFileSecretManger
-    ev_loop: asyncio.AbstractEventLoop = asyncio.get_event_loop()
+
+    try:
+        ev_loop: asyncio.AbstractEventLoop = asyncio.get_event_loop()
+    except Exception:
```
Comment (TheHolyRoger) [inline bin/hummingbot.py:84]: We should perhaps except on the particular exception here and in the quickstart one... I think this exception wouldn't actually be triggered on python 3.10 which we're using now, it's "future-proof" code for 3.11+

File: bin/hummingbot_quickstart.py [hunk 4]
```diff
@@ -148,7 +148,13 @@ def main():
     else:
         secrets_manager = secrets_manager_cls(args.config_password)
 
-    asyncio.get_event_loop().run_until_complete(quick_start(args, secrets_manager))
+    try:
+        ev_loop: asyncio.AbstractEventLoop = asyncio.get_event_loop()
+    except Exception:
```
Comment (TheHolyRoger) [inline bin/hummingbot_quickstart.py:153]: Same with this exception

Comment (TheHolyRoger) [reply bin/hummingbot_quickstart.py:153]: At the least one of us should probably stick a comment in the code explaining the reason I guess

**Comment by cardosofede** [comment]
As this PR is a little bit messy, I wrapped up the changes in this new PR: #6421 
Thanks, @TheHolyRoger for the contributions to this PR and proposals for the excluded test.
@fengtality I talked with THR and he would like to fix the issues of the test that we exclude so we can create a bounty of 200k HBOT, that will cover the changes needed for the test and the work that also THR did here.

**Comment by fengtality** [comment]
> As this PR is a little bit messy, I wrapped up the changes in this new PR: #6421 Thanks, @TheHolyRoger for the contributions to this PR and proposals for the excluded test. @fengtality I talked with THR and he would like to fix the issues of the test that we exclude so we can create a bounty of 200k HBOT, that will cover the changes needed for the test and the work that also THR did here.

Yes, I think it's definitely worthwhile to create the bounty, as it's necessary work. This should come out of the Epoch 5 dev budget that is being proposed: https://snapshot.org/#/hbot.eth/proposal/0xe2dd81f481f6bd84d207a7a4453c5ae820a5c2b64dab63ae746f8ce38a47a434

**Comment by nikspz** [comment]
Tests performed:
Cloned and installed feature branch
Created/started pureMM strategy using binance, ascendex gate.io, kucoin
Created/started perpetualMM using binance perpetual
Review default directional strategy script starting successfully, candles values matched
Created/ started amm arb strategy using Pangolin and Pancake swap, review profitability rates showing
Manually Created docker image successfully

Note: Uniswap ongoing issue https://github.com/hummingbot/hummingbot/issues/6424

Pending:
Docker image multi-arch check

**Review by nikspz — APPROVED** [review]

**Comment by jandykwan** [comment]
> Works on Mac M2 Pro using Miniforge3 here - https://github.com/conda-forge/miniforge/releases/latest/download/Miniforge3-MacOSX-arm64.sh
> 
> Screenshot 2023-06-20 at 10 08 22 PM

when I run the command ./install on macbook m2,  errors as following: 

ResolvePackageNotFound:
  - ujson=1.35
  - python=3.8.2
  - numpy-base=1.20.1

how do you install on mac m3 pro? thanks.

**Comment by cardosofede** [comment]
Use the latest dev branch. If you check the environment.yml now we are
using Python 3.10
